### ISSUE TITLE
Failing tests on master

### ISSUE THREAD

**Issue by nikivazou** [ucsd-progsys/liquidhaskell#453]
This commit https://github.com/ucsd-progsys/liquidhaskell/commit/a74ce288b84fb0176066ea2321df6e39577a7a3a broke the following tests

- Data/Vector/Algorithms/Insertion.hs
- Data/Vector/Algorithms/Common.hs
- Data/Vector/Algorithms/Merge.hs
- Data/Vector/Algorithms/Intro.hs
- Data/Vector/Algorithms/AmericanFlag.hs
- Data/Vector/Algorithms/Radix.hs
- Data/Vector/Algorithms/Search.hs
- Data/Vector/Algorithms/Heap.hs

See [Travis](https://travis-ci.org/ucsd-progsys/liquidhaskell/jobs/79333768) for details

**Comment by ranjitjhala** [comment]
Oddly,

```

$ stack exec -- liquid -i benchmarks/vector-algorithms-0.5.4.2/
benchmarks/vector-algorithms-0.5.4.2/Data/Vector/Algorithms/Insertion.hs

```

says SAFE locally, on my machine. Will pull and rebuild from `master`...


On Friday, September 11, 2015, Niki Vazou <notifications@github.com> wrote:

> Assigned #453 <https://github.com/ucsd-progsys/liquidhaskell/issues/453>
> to @ranjitjhala <https://github.com/ranjitjhala>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucsd-progsys/liquidhaskell/issues/453#event-407617693>
> .
>


-- 
Ranjit.

**Comment by ranjitjhala** [comment]
hmm. I'm having trouble replicating,

stack exec -- liquid -i benchmarks/vector-algorithms-0.5.4.2/
benchmarks/vector-algorithms-0.5.4.2/Data/Vector/Algorithms/Insertion.hs

happily returns SAFE for me, on 'master'.

@nikivazou -- can you replicate this problem?

On Friday, September 11, 2015, Ranjit Jhala <jhala@cs.ucsd.edu> wrote:

> Oddly,
>
> ```
>
> $ stack exec -- liquid -i benchmarks/vector-algorithms-0.5.4.2/
> benchmarks/vector-algorithms-0.5.4.2/Data/Vector/Algorithms/Insertion.hs
>
> ```
>
> says SAFE locally, on my machine. Will pull and rebuild from `master`...
>
>
> On Friday, September 11, 2015, Niki Vazou <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
>
>> Assigned #453 <https://github.com/ucsd-progsys/liquidhaskell/issues/453>
>> to @ranjitjhala <https://github.com/ranjitjhala>.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/ucsd-progsys/liquidhaskell/issues/453#event-407617693>
>> .
>>
>
>
> --
> Ranjit.
>
>

-- 
Ranjit.

**Comment by gridaphobe** [comment]
@ranjitjhala the tests only fail on ghc-7.8.4, so you'll probably need to use 784 to reproduce it locally.

Here are the failure logs: https://travis-ci.org/ucsd-progsys/liquidhaskell/jobs/79333768#L1387-L1403

**Comment by ranjitjhala** [comment]
Ah tis but a single edit with stack...

**Comment by ranjitjhala** [comment]
@nikivazou the issue is this:

```
 LH.GhcMisc.srcSpanSourcePos:0:0: Error: Bad Type Specification
 measure GHC.Types.Eq##6C :: (~# a b c) -> (~ a b c)
     Type constructor GHC.Prim.~# expects a maximum 2 arguments but was given 3 arguments
```

In the particular commit, all top-level data-constructors (including functions) are also declared as `constants` -- the important diff is 

`Language/Haskell/Liquid/Bare/Measure.hs` in `isSimpleType`

but something about the `constant` sort checking in particular `checkTcArity` is complaining about the above. Why is that check failing? In particular, why does it think `GHC.Prim.~# expects a maximum 2 arguments` when it is plainly being applied to 3? Could there be a bug in `realTcArity` ?

**Comment by ranjitjhala** [comment]
Or is this some corner case bug in the GHC 784 API ?

**Comment by nikivazou** [comment]
Looks like at GHC 784 they changed the type of `GHC.Prim.~#` but will look
at it closer tomorrow.

On Fri, Sep 11, 2015 at 8:24 PM, Ranjit Jhala <notifications@github.com>
wrote:

> Or is this some corner case bug in the GHC 784 API ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucsd-progsys/liquidhaskell/issues/453#issuecomment-139705916>
> .
>



-- 
Niki Vazou

**Comment by nikivazou** [comment]
The third argument applied to `GHC.Prim.~#` is a kind variable. I pushed a fix at #454  that just ignores all kinds and waiting for Travis to see if anything breaks.

**Comment by nikivazou** [comment]
Closed with #454

**Comment by ranjitjhala** [comment]
Any idea why this works fine on 7.10 ?

On Monday, September 14, 2015, Niki Vazou <notifications@github.com> wrote:

> Closed with #454 <https://github.com/ucsd-progsys/liquidhaskell/pull/454>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ucsd-progsys/liquidhaskell/issues/453#issuecomment-139997131>
> .
>


-- 
Ranjit.

**Comment by nikivazou** [comment]
I believe because the kind variable was not explicit at 7.10.

Now they are explicitly abstracting and instantiating over kinds.
